generators/app/templates/src/main/webapp/ts/_elasticsearch-reindex.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+<%_ if (enableTranslation) { _%>
+import { JhiLanguageService } from 'ng-jhipster';
+<%_ } _%>
+
+import { ElasticsearchReindexService } from './elasticsearch-reindex.service';
+import { ElasticsearchReindexModalComponent } from './elasticsearch-reindex-modal.component';
+
+@Component({
+    selector: 'jhi-elasticsearch-reindex',
+    templateUrl: './elasticsearch-reindex.component.html'
+})
+export class ElasticsearchReindexComponent {
+
+    constructor(
+        <%_ if (enableTranslation) { _%>
+        private jhiLanguageService: JhiLanguageService,
+        <%_ } _%>
+        private modalService: NgbModal,
+        private elasticsearchReindexService: ElasticsearchReindexService
+    ) {
+        <%_ if (enableTranslation) { _%>
+        this.jhiLanguageService.setLocations(['elasticsearch-reindex']);
+        <%_ } _%>
+    }
+
+    showConfirm() {
+        const modalRef = this.modalService.open(ElasticsearchReindexModalComponent);
+        modalRef.result.then((result) => {
```

Comment:
Oh, okay. I have a feeling that it's intended as a placeholder if the JH user wants to do something with the result. Maybe it would be better to trigger the reindexing here instead of the modal itself?